test/functional/wallet_balance.py

```diff
@@ -140,6 +152,48 @@ def run_test(self):
         after = self.nodes[1].getunconfirmedbalance()
         assert_equal(before + Decimal('0.1'), after)
 
+        # Create 3 more wallet txs, where the last is not accepted to the
+        # mempool because it is the third descendant of the tx above
+        for _ in range(3):
+            # Set amount high enough such that all coins are spent by each tx
+            txid = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 99)
+        assert txid not in self.nodes[0].getrawmempool()
+        assert_equal(self.nodes[0].getbalance(minconf=0), 0)  # wallet txs not in the mempool are untrusted
```

Comment:
A few additional tests you could do here:
- check that `gettransaction(txid)` returns something but `getrawtransaction(txid)` doesn't.
- check that `getrawtransaction()` returns something for the second transaction that was sent (to sanity-check your testing methodology)
- check that `gettransaction(txid)['trusted'] == False`

Reply:
Good suggestions